server/sonar-server/src/main/java/org/sonar/server/user/ws/CurrentAction.java

```diff
@@ -126,17 +130,20 @@ private static void writeGroups(JsonWriter json, Collection<String> groups) {
     json.endArray();
   }
 
-  private static void writePermissions(JsonWriter json, UserSession session) {
+  private void writePermissions(JsonWriter json) {
     json.name("permissions").beginObject();
-    writeGlobalPermissions(json, session);
+    writeGlobalPermissions(json);
     json.endObject();
   }
 
-  private static void writeGlobalPermissions(JsonWriter json, UserSession session) {
+  private void writeGlobalPermissions(JsonWriter json) {
     json.name("global").beginArray();
-    for (String permission : session.globalPermissions()) {
-      json.value(permission);
-    }
+
+    String defaultOrganizationUuid = defaultOrganizationProvider.get().getUuid();
+    GlobalPermissions.ALL.stream()
+      .filter(permission -> userSession.hasOrganizationPermission(defaultOrganizationUuid, permission))
+      .forEach(permission -> json.value(permission));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1612)